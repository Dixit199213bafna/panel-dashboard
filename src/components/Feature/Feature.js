import React from 'react';
import classes from './Feature.module.css';
import Form from 'react-bootstrap/Form'
const feature = (props) => {
    return (
        <div className={props.accordionHeader ? classes.FeatureAccordionHeader : classes.Feature}>
            <label>
                <span className={classes.GreyBox}></span>
                <span>{props.featureName}</span>
            </label>
            <div className={classes.ToggleWithMax}>
            {props.valueList ? <select aria-label={props.name} className={classes.Select}  value={props.value} onChange={(e) => props.onValueChange({
                panel: props.panel,
                featureId: props.featureId,
                value: e.currentTarget.value
            })}>
                {props.valueList.map((val) => <option key={val} value={val}>{val}</option>)}
            </select> : null
            }
            <Form.Check
                type="switch"
                id={props.featureId}  
                checked={props.toggle}
                onChange={(e) => props.onToggleClick({
                    panel: props.panel,
                    featureId: props.featureId,
                    value: e.currentTarget.checked
                })}
            />
            </div>
        </div>
    )
}

export default feature;