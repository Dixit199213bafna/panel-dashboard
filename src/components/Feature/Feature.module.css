.Feature {
    padding: 1rem;
    border: 1px solid #ccc;
    display: flex;
    flex-direction: row;
    justify-content:space-between;
    font-size: initial;
}

.FeatureAccordionHeader {
    width: 100%;
    display: flex;
    justify-content: space-between;
}

.Select {
    background: #282c34;
    color: white;
    border-radius: 10px;
    margin-right: 10px;
}

.ToggleWithMax {
    display: flex;
}

.GreyBox {
    background: #ccc;
    height: 20px;
    width: 20px;
    display: inline-flex;
    top: 4px;
    position: relative;
    right: 6px;
}