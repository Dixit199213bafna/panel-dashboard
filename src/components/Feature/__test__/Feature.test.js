import React from 'react';
import renderer from 'react-test-renderer';
import Feature from '../Feature';

describe('FeatureList', () => {
  let tree;
  const props = {
    featureName: 'ALERT RULES',
    toggle: false,
    valueList: [5,10,20,50],
    value: 10,
    featureId: 32
}
  const component = renderer.create(
    <Feature {...props}/>,
  );
  const instance = component.getInstance();

  it('Feature Initialized', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

});