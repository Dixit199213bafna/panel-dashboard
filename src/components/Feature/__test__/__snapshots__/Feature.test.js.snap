// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList Feature Initialized 1`] = `
<div
  className="Feature"
>
  <label>
    <span
      className="GreyBox"
    />
    <span>
      ALERT RULES
    </span>
  </label>
  <div
    className="ToggleWithMax"
  >
    <select
      className="Select"
      onChange={[Function]}
      value={10}
    >
      <option
        value={5}
      >
        5
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={20}
      >
        20
      </option>
      <option
        value={50}
      >
        50
      </option>
    </select>
    <div
      className="form-switch"
    >
      <input
        checked={false}
        className="form-check-input"
        disabled={false}
        id={32}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
</div>
`;
